spec/services/access_request_approval_service_spec.rb

```diff
@@ -130,5 +130,11 @@
         end
       end
     end
+
+    context "writing data concurrently" do
+      it "writes to both user_permission and organisation_user" do
+        expect { subject }.to change { OrganisationUser.count }.by(1).and change { UserPermission.count }.by(access_request.requester.organisations.flat_map(&:providers).count)
+      end
+    end
```

Comment:
General comment about this spec — this might be a good opportunity to replace `subject` with something more descriptive like `call_service!`. Makes it a bit easier to follow the spec without scrolling back up to understand exactly what `subject` is.